### PR TITLE
[ EBS - 132] - WEB | Convert Row component to Typescript

### PR DESCRIPTION
**Pull request by hamzahsn**
`Description:`

This PR is about converting Row component to typescript

`AC:`

out of scope:

`Jira Link:`

https://uvitadev.atlassian.net/browse/EBS-132

`Screenshot:`

### CODE REVIEW

File: src/Grid/Row/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  column?: boolean;
+  fullWidth?: boolean;
+};
+
+const Row: FC<RowProps> = ({
+  children,
+  style = {},
+  space = "normal",
+  position = "start",
+  positionMd = null,
```
Comment (a7madgamal) [inline src/Grid/Row/index.tsx:28]: does it need to be `null`? what happens if we just don't set a value?

Comment (hamzahsn) [reply src/Grid/Row/index.tsx:28]: but is it a null value though

Comment (a7madgamal) [reply src/Grid/Row/index.tsx:28]: if adding null adds no value please remove it because thats what `undefined` is for and in that case not setting any value is just simpler and less confusing

Comment (hamzahsn) [reply src/Grid/Row/index.tsx:28]: alright Sir 😉

File: package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
   ],
   "scripts": {
     "linter:fix": "prettier --write '{,!(node_modules|.git|storybook-static)/**/}*.{js,json,jsx,md}'",
-    "linter:check": "eslint src/**/* && eslint stories/*.jsx",
+    "linter:check": "eslint 'src/**/*.{tsx, jsx}' && eslint stories/*.jsx",
```
Comment (Weetbix) [inline package.json:19]: Can you explain this change?

It appears to remove linting for .js and .ts files

File: src/Grid/Row/index.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React, { FC, CSSProperties, ReactNode } from "react";
+import Styles from "./index.style";
+import classNames from "../../utils/classNames";
+
+type RowProps = {
+  children: ReactNode;
```
Comment (Weetbix) [inline src/Grid/Row/index.tsx:6]: Children doesnt need to be defined by us as it comes from the react Functional Component prop definition

File: src/Grid/Col/index.js [hunk 4]
```diff
@@ -54,4 +55,5 @@ Col.defaultProps = {
   md: null,
   sm: null,
   xs: null,
+  center: null,
```
Comment (Weetbix) [inline src/Grid/Col/index.js:58]: are these unrelated changes?

File: package.json (hunk 2 above)
Comment (hamzahsn) [reply package.json:19]: remove linting ? 🤔 , so the test was failing cause it was not finding the path for components in the subfolders for `Col` and `Row` and thats why I did it like that, any suggestions ?

File: src/Grid/Row/index.tsx (hunk 3 above)
Comment (hamzahsn) [reply src/Grid/Row/index.tsx:6]: oops missed it 👀

File: src/Grid/Col/index.js (hunk 4 above)
Comment (hamzahsn) [reply src/Grid/Col/index.js:58]: this one was actually cause I have changed the `SideBar.jsx` to `SideBar.tsx` and there was a prop in the Col which is this one but not mentioned in the types so I added it here cause the next ticket is about changing sidebar to `tsx` in both story and component

File: package.json (hunk 2 above)
Comment (Weetbix) [reply package.json:19]: Can you try again? previously this picks up all files and lints what it can including js and ts files. Here you are forcing it to only consider tsx and jsx files

File: src/Grid/Col/index.js (hunk 4 above)
Comment (Weetbix) [reply src/Grid/Col/index.js:58]: Ok so lets remove this change from this PR as its unrelated, you can add it back to the sidebar PR. Otherwise its difficult to review one chunk or work and say its working and makes sense as sits split across multiple PRs with reasonings existing for other unknown reasons

File: package.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
   ],
   "scripts": {
     "linter:fix": "prettier --write '{,!(node_modules|.git|storybook-static)/**/}*.{js,json,jsx,md}'",
-    "linter:check": "eslint src/**/* && eslint stories/*.jsx",
+    "linter:check": "eslint 'src/**/*.{tsx, jsx, ts, js}' && eslint stories/*.jsx",
```
Comment (Weetbix) [inline package.json:19]:
```suggestion
    "linter:check": "eslint 'src/**/*' && eslint stories/*.jsx",
```

File: src/Grid/Col/index.js (hunk 4 above)
Comment (hamzahsn) [reply src/Grid/Col/index.js]: already done in the last commits 😃

File: package.json (hunk 5 above)
Comment (hamzahsn) [reply package.json:19]: this one will keep the error, no ? as it will not detect the files `Col`, `Row`, under Grid 🤔

Comment (Weetbix) [reply package.json:19]: I tried it and it works locally, but it picks up an extra issue with the index file you should look into. 

I think the difference is this includes quotes and passes the handling to eslint, the other one uses bash to expand and includes folders instead of files

File: src/Grid/Col/index.js (hunk 4 above)
Comment (hamzahsn) [reply src/Grid/Col/index.js]: ok, trying it
